pkg/segment/writer/segstore.go

```diff
@@ -96,6 +99,9 @@ type SegStore struct {
 	stbDictEncWorkBuf  [][]string
 	segStatsWorkBuf    []byte
 	SegmentErrors      map[string]*structs.SearchErrorInfo
+	bsPool             []*bitset.BitSet
+	bsPoolCurrIdx      uint32
+	bsPoolCount        uint32
```

Comment:
I'm not sure what the purpose of `bsPoolCount` is. Why not use `len(bsPool)` instead?

Reply:
nit: we might want to move these two a separate struct type that handles all the pooling logic, and just have one field here with that struct. What do you think?

Reply:
good point I remove bsPoolCount and reusing the len(bsPool)